src/Parsers/ParserDropExternalTableQuery.cpp

```diff
@@ -0,0 +1,52 @@
+#include <Parsers/ASTDropQuery.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/ParserDropExternalTableQuery.h>
+
+namespace DB
+{
+
+bool DB::ParserDropExternalTableQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected, [[ maybe_unused ]] bool hint)
+{
+    ParserKeyword s_drop("DROP");
+    ParserKeyword s_external_table("EXTERNAL TABLE");
```

Comment:
Do we need a different `DROP EXTERNAL TABLE` or just DROP ?

Reply:
nI think in general it's a good thing to have a counter part of `CREATE`. The benefits IMO are that: 1. it is more precise, 2. prevent mistakes. For example, `DROP EXTERNAL TABLE foo` will make sure that `foo` is actually an external table, so people won't accidentally drop a stream or view (because of typo, or bad copy/paste).

So, unless there is a strong concern, I will keep this.